core/main.go

```diff
@@ -1,48 +1,62 @@
 package main
 
 import (
-	"log"
 	"path/filepath"
 
 	"github.com/mesg-foundation/core/api"
 	"github.com/mesg-foundation/core/config"
 	"github.com/mesg-foundation/core/database"
 	"github.com/mesg-foundation/core/interface/grpc"
 	"github.com/mesg-foundation/core/logger"
+	"github.com/mesg-foundation/core/service"
 	"github.com/mesg-foundation/core/version"
 	"github.com/mesg-foundation/core/x/xsignal"
 	"github.com/sirupsen/logrus"
 )
 
-func initGRPCServer(c *config.Config) (*grpc.Server, error) {
+type dependencies struct {
+	config      *config.Config
+	serviceDB   database.ServiceDB
+	executionDB database.ExecutionDB
+	api         *api.API
+}
+
+func initDependencies() (*dependencies, error) {
+	dep := dependencies{}
+
+	// init configs.
+	config, err := config.Global()
+	if err != nil {
+		return &dep, err
+	}
+	dep.config = config
+
 	// init services db.
-	db, err := database.NewServiceDB(filepath.Join(c.Core.Path, c.Core.Database.ServiceRelativePath))
+	serviceDB, err := database.NewServiceDB(filepath.Join(dep.config.Core.Path, dep.config.Core.Database.ServiceRelativePath))
 	if err != nil {
-		return nil, err
+		return &dep, err
```

Comment:
not blocking but we can follow returning _nil_ pattern instead of '&dep'

Reply:
done